lms/djangoapps/verify_student/views.py

```diff
@@ -427,6 +444,31 @@ def _redirect_if_necessary(
         if url is not None:
             return redirect(url)
 
+    def _get_verified_modes_for_course(self, course_key):
+        """Retrieve unexpired and expired verified modes for a course.
+
+        Arguments:
+            course_key (CourseKey): The location of the course.
+
+        Returns:
+            Tuple of `(verified_mode, expired_verified_mode)`.  If provided,
```

Comment:
The return value here is a little odd. You get a tuple where at last one value is always None, with a priority on the unexpired verified mode.

Maybe its out of scope for this but it may be cleaner to just have add "is_expired" to the CourseMode model and just return a single verified course mode for the course.